### PR TITLE
Adds a 'dislpay_required' option to TestHarness

### LINKED ISSUE #8700: Add a display option to TestHarness
### Description of the enhancement or error report
The GUI tests require a display for rendering, it is desirable to skip these tests on remote systems that do not have a display.

### Rationale for the enhancement or information for reproducing the error
This will allow us to run python tests on the test machines w/o needing a display.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
None

### PR DESCRIPTION
**Pull request by aeslaughter**
In a quick google this method was recommend, there might be something better.

(closes #8700)

### CODE REVIEW

File: python/TestHarness/testers/Tester.py [hunk 1]
```diff
@@ -358,5 +359,10 @@ def checkRunnableBase(self, options, checks, test_list=None):
                 self.setStatus(reason, self.bucket_skip)
                 return False
 
+        # Check for display
+        if self.specs['display_required'] and os.environ['DISPLAY'] is None:
```
Comment (brianmoose) [inline python/TestHarness/testers/Tester.py]: Do macs always set the `DISPLAY` variable? It is set on mine but that might just be due to having Xquartz installed. We generally want these to run on macs.

Comment (permcody) [inline python/TestHarness/testers/Tester.py]: Right, does this throw a key error if it's not there? Probably should make sure that's handled.

File: python/TestHarness/testers/Tester.py [hunk 2]
```diff
@@ -50,9 +50,10 @@ def validParams():
         params.addParam('asio',          ['ALL'], "A test that runs only if ASIO is available ('ALL', 'TRUE', 'FALSE')")
         params.addParam('depend_files',  [], "A test that only runs if all depend files exist (files listed are expected to be relative to the base directory, not the test directory")
         params.addParam('env_vars',      [], "A test that only runs if all the environment variables listed exist")
-        params.addParam('should_execute', True, 'Whether or not the executeable needs to be run.  Use this to chain together multiple tests based off of one executeable invocation')
+        params.addParam('should_execute', True, 'Whether or not the executable needs to be run.  Use this to chain together multiple tests based off of one executeable invocation')
         params.addParam('required_submodule', [], "A list of initialized submodules for which this test requires.")
         params.addParam('check_input',    False, "Check for correct input file syntax")
+        params.addParam('display_required', False "The test requires and active display for rendering (i.e., ImageDiff tests).")
```
Comment (brianmoose) [inline python/TestHarness/testers/Tester.py]: Need a , after the False

**Review by brianmoose — CHANGES_REQUESTED** [review]

**Comment by permcody** [comment]
Wow... Talk about pulling a @dschwen. Did you even try your code out? 😄 
Maybe you should add a test 😉

**Comment by aeslaughter** [comment]
I just wanted to someone to look at my approach to see if I was even on the correct path.

**Comment by aeslaughter** [comment]
* Adds an option to the test harness to require a display (with a test)
* Remove the --testharness-unittest option and adds the TestHarness test to the python run_tests
* Re-organizes the TestHarness test into individual files so they run faster

File: test/tests/test_harness/display_required [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [./display_required]
+    # This is a dummy test that will always get skipped, see test_HarnessTester
```
Comment (permcody) [inline test/tests/test_harness/display_required]: This comment doesn't really make sense here.

**Review by permcody — APPROVED** [review]
Looks ok

**Review by brianmoose — APPROVED** [review]

**Pull request merged**